File: common-agent/src/main/java/com/dev/lib/harness/session/SessionManager.java
package com.dev.lib.harness.session;

import com.dev.lib.harness.session.model.SpawnArgs;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

public class SessionManager {

    private static final Map<String, AgentSession> SESSION_MAP = new ConcurrentHashMap<>();

    public AgentSession get(String id) {

        return SESSION_MAP.computeIfAbsent(id, it -> new AgentSession(
                SpawnArgs.builder()

                        .build()
        ));
    }

    public void stop() {

    }

}

File: common-agent/src/main/kotlin/com/dev/lib/harness/session/SessionManager.kt
package com.dev.lib.harness.session

import com.dev.lib.harness.sdk.model.ModelProvider
import com.dev.lib.harness.sdk.skill.SkillManager
import org.springframework.stereotype.Component

class SessionManager(
    val sessionStorage: SessionStorage,
    val modelProvider: ModelProvider,
    val skillManager: SkillManager,
) {

    fun openSession(id: String): AgentSession {

        return AgentSession.newBuilder()
            .modelProvider(modelProvider)
            .build()
    }

}